app/scripts/metamask-controller.js

```diff
@@ -4723,7 +4732,10 @@ export default class MetamaskController extends EventEmitter {
     }
   };
 
-  async securityProviderRequest(requestData, methodName) {
+  // We can't code-fence the whole method as some controllers rely on the method. But we can return null
+  // if build is flask
```

Comment:
I think that we can code-fence the entire `securityProviderRequest`. Currently it is used in:
* (TransactionController)[https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/transactions/index.js#L1544] -> it checks if `securityProviderRequest` is defined before calling it.
* SignatureController -> it is optional (see (here)[https://github.com/MetaMask/core/blob/main/packages/signature-controller/src/SignatureController.ts#L119C3-L119C26]). It passes the optional param to the AbstractMessageManager and there it checks if `securityProviderRequest` is defined before calling it (see (here)[https://github.com/MetaMask/core/blob/main/packages/message-manager/src/AbstractMessageManager.ts#L167]).
* MMIController - method does not seem to be used anywhere. Might be worth asking @zone-live about it
* createRPCMethodTrackingMiddleware -> (this)[https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/createRPCMethodTrackingMiddleware.js#L198] is the only place that would require changes.. but I believe that we also need to apply codefencing there as well.

I don't think I have missed any usage.. but might be worth give it a second look.

Reply:
The major issue is the MMIController one I guess. The whole MMI code is behind a code fence already. So can't add another code fence inline.